frontend/src/App.tsx

```diff
@@ -86,40 +174,61 @@ function App(): JSX.Element {
   }, []);
 
   return (
-    <div className="h-screen w-screen flex flex-col">
-      <div className="flex grow bg-neutral-900 text-white min-h-0">
-        <Container
-          orientation={Orientation.HORIZONTAL}
-          className="grow h-full min-h-0 min-w-0 px-3 pt-3"
-          initialSize={500}
-          firstChild={<ChatInterface />}
-          firstClassName="min-w-[500px] rounded-xl overflow-hidden border border-neutral-600"
-          secondChild={
-            <Container
-              orientation={Orientation.VERTICAL}
-              className="grow h-full min-h-0 min-w-0"
-              initialSize={window.innerHeight - 300}
-              firstChild={<Workspace />}
-              firstClassName="min-h-72 rounded-xl border border-neutral-600 bg-neutral-800 flex flex-col overflow-hidden"
-              secondChild={<Terminal />}
-              secondClassName="min-h-72 rounded-xl border border-neutral-600 bg-neutral-800"
-            />
+    <BrowserRouter>
+      <Routes>
+        <Route path="*" element={<Navigate to="/" />} />
+        <Route path="/auth/callback" element={<OAuthPopup />} />
+        <Route
+          path="/"
+          element={
```

Comment:
Best to outsource this element into its own component and reference it here rather than cause `App.tsx` to become too large.